test/instrumentation/index.test.js

```diff
@@ -304,20 +210,6 @@ test('errors should have a transaction id - non-ended transaction', function (t)
   agent.captureError(new Error('bar'))
 })
 
-test('errors should have a transaction id - ended transaction', function (t) {
-  resetAgent(2, function (data) {
-    t.strictEqual(data.transactions.length, 1)
-    t.strictEqual(data.errors.length, 1)
-    const trans = data.transactions[0]
-    t.strictEqual(data.errors[0].transaction_id, trans.id)
-    t.strictEqual(typeof data.errors[0].transaction_id, 'string')
-    t.end()
-  })
-  agent.captureError = origCaptureError
-  agent.startTransaction('foo').end()
-  agent.captureError(new Error('bar'))
-})
-
```

Comment:
REVIEW NOTE: This is an intentional behaviour change. Before this PR an ended transaction would linger as `agent.currentTransaction`. Not any longer.  This test was added in https://github.com/elastic/apm-agent-nodejs/issues/147. My read of that is that there is no need to associate an error with a transaction if it is captured *after* the transaction has ended.